interoperator/controllers/provisioners/sfserviceinstance/sfserviceinstance_controller_test.go

```diff
@@ -331,6 +331,167 @@ func TestReconcileSFServiceInstance_handleError(t *testing.T) {
+            Name:      "instance-id",
+            Namespace: constants.InteroperatorNamespace,
+            Labels: map[string]string{
+                "state":                 "in_queue",
+                constants.ErrorCountKey: "10",
+            },
+        },
+        Spec: osbv1alpha1.SFServiceInstanceSpec{
+            ServiceID:        "service-id",
+            PlanID:           "plan-id",
+            RawContext:       nil,
+            OrganizationGUID: "organization-guid",
+            SpaceGUID:        "space-guid",
+            RawParameters:    nil,
+            PreviousValues:   nil,
+        },
+        Status: osbv1alpha1.SFServiceInstanceStatus{
+            State: "in_queue",
+        },
+    }
+
+    instance2 := &osbv1alpha1.SFServiceInstance{
+        ObjectMeta: metav1.ObjectMeta{
+            Name:      "instance-id-2",
+            Namespace: constants.InteroperatorNamespace,
+            Labels: map[string]string{
+                "state":                 "in_queue",
+                constants.ErrorCountKey: "10",
+            },
+        },
+        Spec: osbv1alpha1.SFServiceInstanceSpec{
+        ServiceID:        "service-id",
+        PlanID:           "plan-id",
+        RawContext:       nil,
+            OrganizationGUID: "organization-guid",
+            SpaceGUID:        "space-guid",
+            RawParameters:    nil,
+            PreviousValues:   nil,
+        },
+        Status: osbv1alpha1.SFServiceInstanceStatus{
+            State: "in_queue",
+        },
+    }
+    err = c.Create(context.TODO(), instance)
+    g.Expect(err).NotTo(gomega.HaveOccurred())
+
+	err = c.Create(context.TODO(), instance2)
+	g.Expect(err).NotTo(gomega.HaveOccurred())
+	serviceInstance := &osbv1alpha1.SFServiceInstance{}
+	g.Eventually(func() error {
+		return c.Get(context.TODO(), instanceKey, serviceInstance)
+	}, timeout).Should(gomega.Succeed())
+
+	r := &ReconcileSFServiceInstance{
+		Client:          c,
+		Log:             ctrlrun.Log.WithName("provisioners").WithName("instance"),
+		scheme:          mgr.GetScheme(),
+		clusterRegistry: mockClusterRegistry,
+		resourceManager: mockResourceManager,
+	}
+	type args struct {
+		object        *osbv1alpha1.SFServiceInstance
+		result        reconcile.Result
+		inputErr      error
+		lastOperation string
+		retryCount    int
+		err           error
+	}
+	tests := []struct {
+		name    string
+		setup   func()
+		args    args
+		want    reconcile.Result
+		wantErr bool
+		wantErrMsg string
+	}{
+        {
+            name: "return default error message if inputErr is empty",
+            args: args{
+                object:        instance,
+                result:        reconcile.Result{},
+                err:           fmt.Errorf("Object 'apiVersion' is missing in 'unstructured object has no version'"),
+                inputErr:      errors.NewMarshalError("Object 'apiVersion' is missing in 'unstructured object has no version'", err),
```

Comment:
Hi @Pooja-08 , probably add a comment here stating a default test on APIVersion Notfound error is being used to test the InputErr not-null case. Helps us when we take a look at the code at later point in time.